src/dso_api/dynamic_api/routers.py

```diff
@@ -250,6 +262,35 @@ def _build_openapi_views(self, datasets: Iterable[Dataset]) -> List[URLPattern]:
             )
         return results
 
+    def _build_index_views(self, datasets: Iterable[Dataset]) -> List[URLPattern]:
+        """Build index views for each sub path
+        Datasets can be grouped on subpaths. This generates a view
+        on each sub path with an index of datasets on that path.
+        """
+
+        # List unique sub paths in all datasets
+        paths = []
+        for ds in datasets:
+            path_components = ds.path.split("/")
+            ds_paths = [
+                "/".join(path_components[0 : i + 1]) for i, x in enumerate(path_components)
+            ]
+            paths += ds_paths[0:-1]
+        paths = set(paths)
+
+        # Create an index view for each path
+        results = []
+        for p in paths:
+            name = p.split("/")[-1].capitalize() + " Datasets"
+            results.append(
+                path(
+                    p + "/",
```

Comment:
What if multiple datasets are using same subpath? How this will look?

Reply:
Like the index page at root, but with only the datasets on that path

Reply:
This is still not covered by tests.

Reply:
Doesnt the new test in  src/tests/test_openapi.py cover this? It has two datasets on the same subpath and checks both are listed.

Reply:
Oh, it does, missed.